Add C9 template (#215)
aws-samples/cfn101-workshop: content/prerequisites/cloud9/index.md

---
title: "AWS Cloud9 Setup (Recommended)"
weight: 200
---

_Setup Duration: ~10 minutes_

## Overview

[AWS Cloud9](https://aws.amazon.com/cloud9/) is a cloud-based IDE that lets you write, run, and debug your code with
just a browser. It includes a code editor, debugger, and terminal. Since your Cloud9 IDE is cloud-based, you can do labs
from your office, home, or anywhere using an internet-connected machine.

> For the best experience with the workshop and minimal setup effort we recommend using it to run this workshop because it

Reviewer: ```suggestion
For the best experience and minimal setup effort, we recommend using it to run this workshop because it
```

:::code{language=shell showLineNumbers=false showCopyAction=true}
git clone https://github.com/aws-samples/cfn101-workshop
:::

#### Open workshop files
The templates and code you will work on is located in the **code** folder.
Expand the tree on the left to open the **code** folder:

![toggletree-png](/static/prerequisites/cloud9/toggletree.png)

### Create a Cloud9 instance via AWS CloudFormation

1. Download the AWS Cloud9 CloudFormation template from workshop repository:
   :::code{language=shell showLineNumbers=false showCopyAction=true}
   curl https://raw.githubusercontent.com/aws-samples/cfn101-workshop/main/code/solutions/cloud9.yaml -o cloud9.yaml
   :::
1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** link in a new tab and log in to your AWS account.
1. Click on **Create stack** (_With new resources (Standard)_ if you have clicked in the top right corner).
1. In **Prepare template**, choose **Template is ready**.
1. In **Template source**, choose **Upload a template file**.
1. Click on **Choose file** button and navigate to the directory you have downloaded the file ins step 1.
1. Select the `cloud9.yaml` file.

> 1. Click **Next**.

Reviewer: ```suggestion
1. Choose **Next**.
```

1. Provide a **Stack name**. For example **cfn-workshop-c9**.
   + The _Stack name_ identifies the stack. Use a name to help you distinguish the purpose of this stack.

>    + Click **Next**.

Reviewer: ```suggestion
   + Choose **Next**.
```

>    + Click **Next**.

Reviewer 2: I think this is an additional **Next** that is not required

> 1. You can leave all the **Parameters** default, click **Next**.

Reviewer: ```suggestion
1. Accept all the default values for **Parameters**, and choose **Next**.
```

> 1. Leave the **Configure stack options** default, click **Next**.

Reviewer: ```suggestion
1. Accept all the default values for **Configure stack options**, and choose **Next**.
```

> 1. On the **Review <stack_name>** page, scroll down to the bottom and tick **Capabilities** check box, click **Submit**.

Reviewer: ```suggestion
1. On the **Review <stack_name>** page, scroll down to the bottom, review and select the **Capabilities** check box, and then choose **Submit**.
```

> 1. You can select the **refresh** button a few times until you see in the status **CREATE_COMPLETE**.

Reviewer: ```suggestion
1. Use the **refresh** button to update the page, and wait until you see the stack to be in the **CREATE_COMPLETE** status.
```

> 1. Navigate to the **Outputs** and select the URL provided to open your Cloud9 environment.

Reviewer: ```suggestion
1. Navigate to the **Outputs** pane, and follow the provided URL to open your Cloud9 environment.
```

1. In a Cloud9 terminal, download and run a bootstrap script to upgrade SAM CLI to the most recent version and install few other tools:
    :::code{language=shell showLineNumbers=false showCopyAction=true}
    curl 'https://static.us-east-1.prod.workshops.aws/public/ca3cac3e-84b1-4b6a-bd2b-d4565df5a5db/static/bin/bootstrap.sh' | bash
    :::
1. Verify the new version
    :::code{language=shell showLineNumbers=false showCopyAction=true}
    sam --version
    :::

#### Open workshop files

> The templates and code you will work on is located in the **code** folder.

Reviewer: ```suggestion
The templates and the code you will work on are both located in the **code** folder.
```
